internal/state/execution.go

```diff
@@ -416,43 +414,51 @@ func buildLastCommitInfo(block *types.Block, store Store, initialHeight int64) a
 	}
 }
 
-// extendedCommitInfo expects a CommitInfo struct along with all of the
-// original votes relating to that commit, including their vote extensions. The
-// order of votes does not matter.
-func extendedCommitInfo(c abci.CommitInfo, votes []*types.Vote) abci.ExtendedCommitInfo {
-	if len(c.Votes) != len(votes) {
-		panic(fmt.Sprintf("extendedCommitInfo: number of votes from commit differ from the number of votes supplied (%d != %d)", len(c.Votes), len(votes)))
-	}
-	votesByVal := make(map[string]*types.Vote)
-	for _, vote := range votes {
-		if vote != nil {
-			valAddr := vote.ValidatorAddress.String()
-			if _, ok := votesByVal[valAddr]; ok {
-				panic(fmt.Sprintf("extendedCommitInfo: found duplicate vote for validator with address %s", valAddr))
-			}
-			votesByVal[valAddr] = vote
-		}
+// buildLastExtendedCommitInfo expects an ExtendedCommitInfo, which includes
+// the original precommit votes along with their vote extensions from the last
+// commit. It populates the corresponding ABCI protobuf structures using the
+// validator set obtained from the store at the height of the last extended
+// commit.
+func buildLastExtendedCommitInfo(ec *types.ExtendedCommit, store Store, initialHeight int64) abci.ExtendedCommitInfo {
+	// We assume the current height is ec.Height + 1.
```

Comment:
This makes sense as an assumption, but I'm not sure how it relates to the test that follows—is `initialHeight` what we mean by "current height" in this context?

Reply:
`initialHeight` is the first height of the chain.
It can be configured in the genesis file:
https://github.com/tendermint/tendermint/blob/b1896ebca6a857c9ec97e8ec086e7ab5d2f0fbf5/types/genesis.go#L71
For brand new chains, this will take the [default value of 1](https://github.com/tendermint/tendermint/blob/b1896ebca6a857c9ec97e8ec086e7ab5d2f0fbf5/types/genesis.go#L111), but for chains that are aimed to be the successor to a previous chain, say after a hard fork upgrade, this value will be set to be one beyond the last height of the previous chain.